h2/src/main/org/h2/value/Transfer.java

```diff
@@ -494,6 +494,20 @@ public void writeValue(Value v) throws IOException {
             }
             break;
         }
+        case Value.ENUM: {
+            ValueEnum ve = (ValueEnum) v;
+            String[] enumerators = ve.getEnumerators();
+
+            writeInt(enumerators.length);
+
+            for (String member : enumerators) {
+                writeString(member);
+            }
+
+            writeInt(ve.getInt());
+
+            break;
+        }
```

Comment:
OK @grandinj makes sense. However, in this case I think I will also have to pass along some identifier so that `readValue` can find the complete set of enumerators via a `DataHandler` (or similar) interface? Let me know if that's not right.